src/ast/CMakeLists.txt

```diff
@@ -100,9 +100,17 @@ if(STATIC_LINKING)
   target_link_libraries(ast PUBLIC ${clang_libs})
   target_link_libraries(ast PUBLIC ${llvm_libs})
 else()
-  # llvm_config macro comes from the LLVM toolchain and will auto-resolve component
-  # names to library names. USE_SHARED option will tell llvm_config macro to prefer
-  # shared library / DLL on the system over the static libraries
-  llvm_config(ast USE_SHARED bpfcodegen ipo irreader mcjit orcjit)
+  find_package(LLVM QUIET CONFIG)
+  if (TARGET LLVM)
```

Comment:
What's the significance of this check? I presume it means the config mode search succeeded, but that wouldn't explain why it's still ok to use `llvm_map_components_to_libnames()` and `llvm_expand_dependencies()` in the failure branch below, as they're defined in the same file as `llvm_config()`.

Reply:
The `llvm_config()` links the LLVM lib which causes the below error when I build bpftrace for Android.
```
ld.lld: error: unable to find library -lLLVM
```
code ref[0]:
```
target_link_libraries(${executable} PRIVATE LLVM)
```
`llvm_map_components_to_libnames ` and `llvm_expand_dependencies` are old code before https://github.com/bpftrace/bpftrace/pull/3580, which works fine.
![image](https://github.com/user-attachments/assets/26d48d88-20ef-4a1e-aba5-ce78b473ed7a)


[0] https://github.com/llvm/llvm-project/blob/7a8fe0f83c4ba03d07aef9243596d67af74a3b87/llvm/cmake/modules/LLVM-Config.cmake#L95

Reply:
Ah ok, so your build environment doesn't have libLLVM.so. And the use of `llvm_config(.. USE_SHARED ..)` is forcing linking to the libLLVM.so. The `if (TARGET LLVM)` check seems more correct. How about dropping the `USE_SHARED` arg in the else branch instead of bringing back the old logic?